### PR TITLE
Networking basics

### PR DESCRIPTION
**Pull request by Atomic182**
Basic networking classes.

### CODE REVIEW

File: src/networking/sock/Socket.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+#include "Socket.h"
+
+namespace ArgosNet
+{
+
+Socket::Socket() {}
+
+Socket::~Socket() {}
+
+bool Socket::Open(unsigned short port)
```
Comment (travisdoor) [inline src/networking/sock/Socket.cpp]: Metody malym?

File: src/networking/sock/Socket.h [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+//
+
+#ifndef ARGOSRTS_SOCKET_H
+#define ARGOSRTS_SOCKET_H
+
+#define PLATFORM_WINDOWS 1
+#define PLATFORM_MAC 2
+#define PLATFORM_UNIX 3
+
+#if defined(_WIN32)
```
Comment (travisdoor) [inline src/networking/sock/Socket.h]: Tohle by se pak hodilo dat nekde globalne.

Comment (Atomic182) [inline src/networking/sock/Socket.h]: Nejaky global header pouzivany u OS sensitive casti kodu ?

Comment (travisdoor) [inline src/networking/sock/Socket.h]: jj neco tak

**Pull request merged**